components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml

```diff
@@ -430,6 +430,40 @@ document.observe('xwiki:dom:loaded', function()
   // when date of death years are given as a decade, the value of that field is empty.
   // LiveValidation does not show error message on this case
   date_of_death.__validation.displayMessageWhenEmpty = true;
+
+  // show/hide dates of birth and death depending on life status selection
+  var displayDates = function(life_status) {
+     var birth_date_entered = $$("[id$='date_of_birth_entered']")[0];
+     var death_date_entered = $$("[id$='date_of_death_entered']")[0];
+
+     var date_of_birth_block = $('date-of-birth-block');
+     var date_of_death_block = $('date-of-death-block');
+
+     if (life_status == 'alive') {
+        date_of_birth_block.show();
+        date_of_death_block.hide();
+        date_of_death.value = "";
+        death_date_entered.value="{}"
+     } else if (life_status == 'deceased') {
+        date_of_birth_block.show();
+        date_of_death_block.show();
+     } else {
+        date_of_birth_block.hide();
+        date_of_death_block.hide();
+        date_of_death.value = "";
+        death_date_entered.value="{}"
+        date_of_birth.value = "";
+        birth_date_entered.value="{}"
+     }
+  }
+
+  var life_status = $$("[id$='life_status']")[0];
+  displayDates(life_status.value);
```

Comment:
This will throw an error if life status is disabled in the patient form. Always check if the element exists before reading any data from it.